ecl2df/common.py

```diff
@@ -627,43 +628,87 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
-    """Return a dictionary from (int) K layers in the simgrid to strings
+def _is_color(input_string: str) -> bool:
+    """Check if the input string is a 6 digit hexadecimal color code"""
+    if input_string.startswith("#") and len(input_string) == 7:
+        return True
+    return False
 
-    Typical usage is to map from grid layer to zone names.
 
-    The layer filename must currently follow format:
+def parse_lyrfile(filename: str) -> Dict[str, Dict[str, Any]]:
+    """Return a dictionary presentation of the lyr file.
 
-      'ZoneA' 1-4
-      'ZoneB' 5-10
+    The lyr file contains data of the following format,
+    where the color code is optional:
 
-    where the single quotes are optional for zones without spaces.
-    Write single layer zones as 11-11. NB: ResInsight requires single
-    quotes always.
+      'ZoneA' 1-4     #FFE5F7
+      'ZoneB' 5-10    #FFE55E
 
-    Either "--" or "#" can be used to denote comments.
+    Only 6 digit color codes starting with # are supported.
+
+    The dictonary representation has the following format:
+
+    {
+        "zoneA:{
+            "from_layer": 1,
+            "to_layer": 4
+            "color": "#FFE5F7"
+        },
+        "zoneB":{
+            "from_layer": 5,
+            "to_layer": 10
+            "color": "#FFFF00"
+        }
+    }
 
     Args:
         filename (str): Absolute path to a zone map file (lyr format)
 
     Returns:
-        dict, integer keys which are the K layers. Every layer mentioned
-            in the interval in the input file is present. Can be empty.
+        Dict[str, Dict[str, Any]], keys: zone name, value: zone attributes
     """
+
     zonelines = Path(filename).read_text().splitlines()
 
     # Remove comments, support both "--" and "#":
     zonelines = [line.split("--")[0].strip() for line in zonelines]
     zonelines = [line for line in zonelines if line and not line.startswith("#")]
 
-    zonemap = {}
+    output: Dict[str, Dict[str, Any]] = {}
+
     for line in zonelines:
         try:
             linesplit = shlex.split(line)
-            (k_0, k_1) = "".join(linesplit[1:]).split("-")
-            zonemap.update(dict.fromkeys(range(int(k_0), int(k_1) + 1), linesplit[0]))
+            zone_name = linesplit[0]
+            zone_color = linesplit.pop(-1) if _is_color(linesplit[-1]) else None
+            (from_layer, to_layer) = "".join(linesplit[1:]).split("-")
+            zone_attributes = {"from_layer": int(from_layer), "to_layer": int(to_layer)}
+            if zone_color is not None:
+                zone_attributes["color"] = zone_color
+            output[zone_name] = zone_attributes
         except ValueError:
-            logger.error("Could not parse zonemapfile %s", filename)
+            logger.error("Could not parse lyr file %s", filename)
             logger.error("Failed on content: %s", line)
             return None
+    return output
+
+
+def convert_lyrparse_to_zonemap(lyrparse: dict) -> Dict[int, str]:
```

Comment:
`lyrparse` -> `lyrdict` ?

Reply:
Yes, I agree. I will change it to that everywhere